dan_layer/consensus/src/hotstuff/on_receive_proposal.rs

```diff
@@ -261,9 +261,14 @@ where TConsensusSpec: ConsensusSpec
                 .await?;
             let vote = self.generate_vote_message(block, decision).await?;
 
-            self.send_vote_to_leader(local_committee, vote, block.height()).await?;
+            self.send_vote_to_leader(local_committee, vote.clone(), block.height())
+                .await?;
 
-            self.store.with_write_tx(|tx| block.as_last_voted().set(tx))?;
+            self.store.with_write_tx(|tx| {
+                block
+                    .as_last_voted(vote.epoch, vote.decision, vote.signature, vote.merkle_proof)
```

Comment:
I think Block::as_last_voted doesnt make sense anymore. What about `VoteMessage::to_last_voted` ? 

```suggestion
                vote.to_last_voted().set(tx)
```

Reply:
hmm, last voted is now getting destroyed with dummy blocks on leader failure trigger. I need to store it somewhere else.

Reply:
Yeah that part of the code is in flux at the moment. The latest code is in #651.